### PR TITLE
ddl: throw warning when `ALTER TABLE ORDER BY` on table with primary key

### PR DESCRIPTION
**Pull request by yuanjize**
### What problem does this PR solve? 
  Fix pingcap/parser#608
  Throw a warning if a table has the primary key when do a `alter order by` ddl.

### What is changed and how it works?

tidb> create table t(id int primary key not null);
tidb> alter table t order by id;
tidb> show warnings;
+---------+------+------------------------------------------------------------------------------+
| Level   | Code | Message                                                                      |
+---------+------+------------------------------------------------------------------------------+
| Warning | 1105 | ORDER BY ignored as there is a user-defined clustered index in the table 't' |
+---------+------+------------------------------------------------------------------------------+

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: ddl/db_test.go [hunk 1]
```diff
@@ -4240,6 +4240,23 @@ func (s *testDBSuite2) TestDDLWithInvalidTableInfo(c *C) {
 	c.Assert(err.Error(), Equals, "[parser:1064]You have an error in your SQL syntax; check the manual that corresponds to your TiDB version for the right syntax to use line 1 column 94 near \"then (b / a) end));\" ")
 }
 
+func (s *testDBSuite1) TestAlterOrderBy(c *C) {
+	s.tk = testkit.NewTestKit(c, s.store)
+	s.tk.MustExec("use " + s.schemaName)
+	s.tk.MustExec("create table ob (pk int primary key, c int default 1, c1 int default 1, KEY cl(cl))")
```
Comment (tangenta) [inline ddl/db_test.go]:
```suggestion
	s.tk.MustExec("create table ob (pk int primary key, c int default 1, c1 int default 1, KEY cl(c1))")
```

File: ddl/db_test.go [hunk 2]
```diff
@@ -4240,6 +4240,23 @@ func (s *testDBSuite2) TestDDLWithInvalidTableInfo(c *C) {
...
+
+	// Test order by with primary key
+	s.tk.MustExec("alter table ob order by c")
+	c.Assert(s.tk.Se.GetSessionVars().StmtCtx.WarningCount(), Equals, uint16(1))
+	s.tk.MustQuery("show warnings").Check(testutil.RowsWithSep("|", "Warning|1105|ORDER BY ignored as there is a user-defined clustered index in the table '%s'", "ob"))
+
+	//Test order by with no primary key
```
Comment (tangenta) [inline ddl/db_test.go]:
```suggestion
	// Test order by with no primary key
```

File: ddl/db_test.go [hunk 3]
```diff
@@ -4240,6 +4240,23 @@ func (s *testDBSuite2) TestDDLWithInvalidTableInfo(c *C) {
...
+	s.tk.MustExec("alter table ob drop primary key")
+	s.tk.MustExec("alter table ob order by c")
+	c.Assert(s.tk.Se.GetSessionVars().StmtCtx.WarningCount(), Equals, uint16(0))
+	s.tk.MustExec("drop table ob")
+}
+
```
Comment (tangenta) [inline ddl/db_test.go]: Currently, TiDB does not support drop primary key unless the option `alter-primary-key` is enabled in the configuration. For simplicity, we can drop the old table and recreate a new one without the primary key.

File: ddl/ddl_api.go [hunk 4]
```diff
@@ -4143,3 +4145,17 @@ func (d *ddl) RepairTable(ctx sessionctx.Context, table *ast.TableName, createSt
...
+	_, tb, err := d.getSchemaAndTableByIdent(ctx, ident)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	for _, item := range tb.Cols() {
+		if item.Flag&mysql.PriKeyFlag != 0 {
+			ctx.GetSessionVars().StmtCtx.AppendWarning(errors.Errorf("ORDER BY ignored as there is a user-defined clustered index in the table '%s'", ident.Name))
+			break
+		}
+	}
```
Comment (tangenta) [inline ddl/ddl_api.go]: Suggested change:
```go
if tb.Meta().GetPkColInfo() != nil {
    // ...
}
```

File: ddl/ddl_api.go [hunk 5]
```diff
@@ -2066,6 +2066,8 @@ func (d *ddl) AlterTable(ctx sessionctx.Context, ident ast.Ident, specs []*ast.A
 			}
 		case ast.AlterTableSetTiFlashReplica:
 			err = d.AlterTableSetTiFlashReplica(ctx, ident, spec.TiFlashReplica)
+		case ast.AlterTableOrderByColumns:
+			err = d.AlterTableOrderBy(ctx, ident)
```
Comment (reafans) [inline ddl/ddl_api.go]:
```suggestion
			err = d.OrderByColumns(ctx, ident)
```

File: ddl/ddl_api.go [hunk 6]
```diff
@@ -4143,3 +4145,14 @@ func (d *ddl) RepairTable(ctx sessionctx.Context, table *ast.TableName, createSt
 	err = d.callHookOnChanged(err)
 	return errors.Trace(err)
 }
+
+func (d *ddl) AlterTableOrderBy(ctx sessionctx.Context, ident ast.Ident) error {
```
Comment (reafans) [inline ddl/ddl_api.go]:
```suggestion
func (d *ddl) OrderByColumns(ctx sessionctx.Context, ident ast.Ident) error {
```

**Comment by Deardrops** [comment]
@yuanjize please change PR title to 
ddl: throw warning when `alter table order by` on table with primary key

**Comment by reafans** [comment]
lgtm

**Pull request merged**